### PR TITLE
GFORMS-1055 - Make the file upload component work so it does not use …

### PR DESCRIPTION
**Pull request by josef-vlach**
…javascript

### CODE REVIEW

**Review by girishkamat — COMMENTED** [review]
We get the following error when no file is chosen. Can this be improved?

`Only one file at a time can be uploaded`

**Comment by josef-vlach** [comment]
In what way? Please read Slack discussion about this for some background.

File: app/uk/gov/hmrc/gform/sharedmodel/form/FormComponentIdToFileIdMapping.scala [hunk 1]
```diff
@@ -39,19 +38,17 @@ case class FormComponentIdToFileIdMapping(mapping: Map[FormComponentId, FileId])
...
+      val inverse: Option[FileId] = inverseMapping.get(fileId).map(_.value).map(FileId.apply)
+      inverse.map { fileId =>
+        loop(fileId).getOrElse(fileId)
+      }
     }
-    FileId(mapping.get(formComponentId).fold(nextAvailable)(_.value))
 
+    FileId(
+      mapping.get(formComponentId).orElse(loop(FileId(formComponentId.value))).fold(formComponentId.value)(_.value)
+    )
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/sharedmodel/form/FormComponentIdToFileIdMapping.scala:51]: Minor: the expression could be simplified

` mapping
      .getOrElse(formComponentId, 
          loop(FileId(formComponentId.value))
            .getOrElse(FileId(formComponentId.value)))`

File: public/javascripts/gformFileUpload.js [hunk 2]
```diff
@@ -259,11 +259,14 @@
           interpolate(strings.maxSizeError[lang], [maxFileSize])
         );
       }
-
-      updateMapping($input.attr("id"), fileId, formTemplateId, accessCode)
+      uploadFile(file, fileId)
         .then(function(response) {
```
Comment (girishkamat) [inline public/javascripts/gformFileUpload.js:263]: why was the order reversed?

**Review by girishkamat — APPROVED** [review]

File: public/javascripts/gformFileUpload.js (hunk 2 above)
Comment (josef-vlach) [reply public/javascripts/gformFileUpload.js:263]: To match order of events in no-js flow.

**Pull request merged**